app/src/main/java/org/oppia/android/app/onboarding/OnboardingFragmentPresenter.kt

```diff
@@ -36,13 +84,197 @@ class OnboardingFragmentPresenter @Inject constructor(
         appLanguageResourceHandler.getStringInLocale(R.string.app_name)
       )
 
-      onboardingLanguageLetsGoButton.setOnClickListener {
-        val intent =
-          OnboardingProfileTypeActivity.createOnboardingProfileTypeActivityIntent(activity)
-        fragment.startActivity(intent)
+      onboardingAppLanguageViewModel.supportedAppLanguagesList.observe(
+        fragment,
+        { languagesList ->
+          val adapter = ArrayAdapter(
+            fragment.requireContext(),
+            R.layout.onboarding_language_dropdown_item,
+            R.id.onboarding_language_text_view,
+            languagesList
+          )
+          onboardingLanguageDropdown.setAdapter(adapter)
+        }
+      )
+
+      onboardingAppLanguageViewModel.languageSelectionLiveData.observe(
+        fragment,
+        { language ->
+          selectedLanguage = language
+          onboardingLanguageDropdown.setText(selectedLanguage, false)
+        }
+      )
+
+      onboardingLanguageDropdown.apply {
+        setRawInputType(EditorInfo.TYPE_NULL)
+
+        onItemClickListener =
+          AdapterView.OnItemClickListener { _, _, position, _ ->
+            adapter.getItem(position).let { selectedItem ->
+              if (selectedItem != null) {
+                selectedLanguage = selectedItem as String
+                onboardingAppLanguageViewModel.setSelectedLanguageDisplayName(selectedLanguage)
+              }
+            }
+          }
       }
+
+      onboardingLanguageLetsGoButton.setOnClickListener { updateSelectedLanguage(selectedLanguage) }
     }
 
     return binding.root
   }
+
+  private fun updateSelectedLanguage(selectedLanguage: String) {
+    val oppiaLanguage = appLanguageResourceHandler.getOppiaLanguageFromDisplayName(selectedLanguage)
+    val selection = AppLanguageSelection.newBuilder().setSelectedLanguage(oppiaLanguage).build()
+    translationController.updateAppLanguage(profileId, selection).toLiveData()
+      .observe(
+        fragment,
+        { result ->
+          when (result) {
+            is AsyncResult.Success -> {
+              val intent =
+                OnboardingProfileTypeActivity.createOnboardingProfileTypeActivityIntent(activity)
+              intent.decorateWithUserProfileId(profileId)
+              fragment.startActivity(intent)
+            }
+            is AsyncResult.Failure -> oppiaLogger.e(
+              "OnboardingFragment",
+              "Failed to set AppLanguageSelection",
+              result.error
+            )
+            is AsyncResult.Pending -> {}
+          }
+        }
+      )
+  }
+
+  private fun getSystemLanguage() {
```

Comment:
Alternative name suggestion: initializeSelectedLanguageToSystemLanguage.

Ditto for ``getSupportedLanguages`` (since these aren't traditional getters).